### PR TITLE
Make ShieldData self contained

### PR DESCRIPTION
**Pull request by panleone**
save and load the extended full viewing key in place of a useless default address in this way:

- `ShieldData` is self contained and we can create a full view instance only from it
- `save` and `load` are no longer async
- On MPW we don't have to save separately `ShieldData` and `extfvk`

### CODE REVIEW

File: js/pivx_shield.js [hunk 1]
```diff
@@ -219,15 +217,11 @@ export class PIVXShield {
...
    * @param {ShieldData} shieldData - shield data
    */
-  async load(shieldData) {
-    const { address, _ } = await this.callWorker(
-      "generate_default_payment_address",
-      this.extfvk,
-      this.isTestNet,
-    );
-    if (address != shieldData.defaultAddress) {
+  load(shieldData) {
```
Comment (Duddino) [inline js/pivx_shield.js:220]: I would make `load` static and make it return a PIVXShield instance instead of adding other things to the create function

**Review by Duddino — CHANGES_REQUESTED** [review]

**Comment by panleone** [comment]
done

File: js/pivx_shield.js [hunk 2]
```diff
@@ -197,42 +190,43 @@ export class PIVXShield {
...
+        isTestNet: this.isTestNet,
       }),
     );
   }
 
   /**
-   * Load shieldWorker from a shieldData
+   * Creates a PIVXShield object from a ShieldData
    * @param {ShieldData} shieldData - shield data
    */
```
Comment (Duddino) [inline js/pivx_shield.js:209]:
```suggestion
   * Creates a PIVXShield object from a ShieldData
   * @param {string} shieldData - shield data
   */
```
Change this to string and add `JSON.parse` inside the function. We are not making shieldData transparent to the user, so I would avoid passing the object directly.

**Review by Duddino — CHANGES_REQUESTED** [review]

File: js/pivx_shield.js (hunk 2 above)
Comment (panleone) [reply js/pivx_shield.js]: done

**Review by Duddino — APPROVED** [review]
utACK

**Pull request merged**